stream/sentinel-intel/src/sentinel_intel_connector/utils.py

```diff
@@ -0,0 +1,188 @@
+    return threat_type
+
+
+def get_description(data: dict) -> str:
+    """
+    Get a description according to observable.
+    :param data: Observable data to extract description from
+    :return: Observable description summary or "No Description"
+    """
+    stix_description = OpenCTIConnectorHelper.get_attribute_in_extension(
+        "description", data
+    )
+    return stix_description[0:99] if stix_description is not None else "No description"
+
+
+def get_action(data: dict) -> str:
+    """
+    Get an action according to observable score.
+    :param data: Observable data to get action from
+    :return: Action name or "unknown"
+    """
+    score = OpenCTIConnectorHelper.get_attribute_in_extension("score", data)
+    if score is None:
+        action = "unknown"
+    elif score >= 50:  # self.config.confidence_levek == 50
+        action = "block"
+    elif score < 50 and score != 0:  # self.config.confidence_levek == 50
+        action = "alert"
+    elif score == 0:
+        action = "allow"
+    else:
+        action = "unknown"
+    return action
+
+
+def get_expiration_datetime(data: dict, expiration_time: int) -> str:
+    """
+    Get an expiration datetime for an observable.
+    :param data: Observable data to calculate expiration with
+    :param expiration_time: Duration after which observable is considered as expired
+    :return: Datetime of observable expiration
+    """
+    updated_at = OpenCTIConnectorHelper.get_attribute_in_extension("updated_at", data)
+    datetime_object = datetime.strptime(updated_at, "%Y-%m-%dT%H:%M:%S.%fZ")
+    age = timedelta(expiration_time)
+    expire_datetime = datetime_object + age
+    expiration_datetime = expire_datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
+    return expiration_datetime
+
+
+def get_tlp_level(data: dict) -> str:
+    """
+    Get a TLP level for an observable.
+    :param data: Observable data to extract TLP level from
+    :return: TLP level or "unknown"
+    """
+    if "marking-definition--5e57c739-391a-4eb3-b6be-7d15ca92d5ed" in str(data):
+        tlp_level = "red"
+    elif "marking-definition--826578e1-40ad-459f-bc73-ede076f81f37" in str(
+        data
+    ) or "marking-definition--f88d31f6-486f-44da-b317-01333bde0b82" in str(data):
+        tlp_level = "amber"
+    elif "marking-definition--34098fce-860f-48ae-8e50-ebd3cc5e41da" in str(data):
+        tlp_level = "green"
+    elif "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9" in str(data):
+        tlp_level = "white"
+    else:
+        tlp_level = "unknown"
+    return tlp_level
+
+
+def get_tags(data: dict) -> list[str]:
+    """
+    Get tags for an observable.
+    :param data: Observable data to extract tags from
+    :return: List of tags
+    """
+    tags = ["opencti"]
+    labels = OpenCTIConnectorHelper.get_attribute_in_extension("labels", data)
+    return tags + labels if labels is not None else tags
+
+
+def get_hash_type(data: dict) -> str:
```

Comment:
```suggestion
def get_hash_type(data: dict) -> str:|None
```
the null return is implicit if data["hashes"] is not handle.

Nitpicking : style is not consistent between your various selectors

Reply:
I updated this file and tried to use the same code style as discussed. Let me know if it's not what you meant :)